src/components/questionnaire/questionnaire-table.tsx

```diff
@@ -18,59 +18,97 @@ import strings from "src/localization/strings";
 import UserRoleUtils from "src/utils/user-role-utils";
 import { DataGrid, type GridRenderCellParams } from "@mui/x-data-grid";
 import { useLambdasApi } from "src/hooks/use-api";
+import { useSetAtom } from "jotai";
+import { errorAtom } from "src/atoms/error";
 
 /**
  * Questionnaire Table Component
  * @returns Questionnaires from DynamoDB rendered in a x-data-grid table
  */
 const QuestionnaireTable = () => {
   const adminMode = UserRoleUtils.adminMode();
-  const { questionnaireApi } = useLambdasApi();
+  const { questionnairesApi } = useLambdasApi();
   const [questionnaires, setQuestionnaires] = useState<Questionnaire[]>([]);
-  const [loading, setLoading] = useState(true);
+  const [loading, setLoading] = useState(false);
   const [dialogOpen, setDialogOpen] = useState(false);
   const [deleteId, setDeleteId] = useState<string | null>(null);
   const [deleteTitle, setDeleteTitle] = useState<string | null>(null);
+  const setError = useSetAtom(errorAtom);
 
   useEffect(() => {
     const fetchQuestionnaires = async () => {
       setLoading(true);
       try {
-        const questionnaires = await questionnaireApi.listQuestionnaires();
+        const questionnaires = await questionnairesApi.listQuestionnaires();
         setQuestionnaires(questionnaires);
-      } finally {
-        setLoading(false);
+      } catch (error) {
+        setError(`${strings.error.questionnaireLoadFailed}, ${error}`);
       }
+      setLoading(false);
     };
     fetchQuestionnaires();
   }, []);
 
+  /**
+   * Function to open the dialog for deleting the questionnaire
+   * @param id - Questionnaire ID
+   * @param title - Questionnaire Title
+   */
   const handleOpenDialog = (id: string, title: string) => {
     setDeleteId(id);
     setDeleteTitle(title);
     setDialogOpen(true);
   };
 
+/**
+ * Function to close the dialog for deleting the questionnaire
+ */
   const handleCloseDialog = () => {
     setDialogOpen(false);
     setDeleteId(null);
     setDeleteTitle(null);
   };
 
+  /**
+   * Function to delete the questionnaire
+   */
   const handleConfirmDelete = async () => {
-    if (!deleteId) return;
+    if (!deleteId) 
+      return;
     setLoading(true);
     try {
-      await questionnaireApi.deleteQuestionnaire({ id: deleteId });
+      await questionnairesApi.deleteQuestionnaires({ id: deleteId });
       setQuestionnaires((prevQuestionnaires: Questionnaire[]) =>
         prevQuestionnaires.filter((questionnaire: Questionnaire) => questionnaire.id !== deleteId)
       );
       handleCloseDialog();
     } catch (error) {
-      console.error("Delete Questionnaire failed:", error);
-    } finally {
-      setLoading(false);
+      setError(`${strings.error.questionnaireDeleteFailed}, ${error}`);
     }
+      setLoading(false);
+    
+  };
+
+  /**
+   * Function to render the confirm delete dialog
+   */
+  const renderConfirmDeleteDialog = () => {
+    return (
```

Comment:
This could be converted to an implicit return. e.g. remove the return () and replace the {} with () (or chat GPT if unsure).